### PR TITLE
Makes description columns non-nullable.

### LINKED ISSUE #9414: Remove use of sql.NullString and ensure all description fields are not nullable
Historically some database tables have allowed NULL description fields.
However there is no need for this as an empty string aligns more closely with Go's default string value and can indicate "no description" without needing the field to be NULLable.

Right now we use a mixture of Go's `sql.NullString{}` to handle parsing NULLs from the database results or handle it at the databse layer using `IFNULL()` in the query.

Requiring these approaches is error prone when refactoring code (see https://github.com/lxc/lxd/issues/9385), because new records are created with descriptions that are an empty string (if not specified) and so the automated tests do not account for older records that have a NULL description.

We should:

- Patch the DB to make the description fields not NULLable.
- Remove the NULL handling in the code for `description` fields.

### PR DESCRIPTION
**Pull request by markylaing**
This reduces error prone handling checks of `sql.NullString`. 

https://github.com/lxc/lxd/pull/9692 Must be merged before this.

Fixes #9414

### CODE REVIEW

**Comment by markylaing** [comment]
@tomponline I will rebase to fix the commits once the migration tests pass. Also, @stgraber mentioned that the re-adding the views may not be necessary as they are not in use. If so, perhaps they can be removed in a separate PR before this one.

**Comment by tomponline** [comment]
@markylaing yep if the views can be removed as not used then defo this should be a separate PR before this one, thanks.

**Comment by tomponline** [comment]
@markylaing if adding the `DEFAULT ''` is creating additional changes to tables that were otherwise untouched I'd leave that outside as the default default is "" anyway (once NOT NULL). So if a description field is already `NOT NULL` then leave it alone for this change, and ones that are nullable just make `NOT NULL` without a default.

**Comment by tomponline** [comment]
A general comment is that each commit should contain the patch that is altering the schema and the output of `make update-schema`. E.g. https://github.com/lxc/lxd/commit/f420487375fa793197f103e3dbb2b1eff6e2ee6e#diff-960311a5bca4cb7bf50b28cb0a3abd5673c4caccf7989fe502f7ffd204ac643a

**Comment by tomponline** [comment]
@markylaing if you're finding that you're having to drop tables because of references to the table you have to remove and re-create to fix the description field, you may find that temporarily disabling foreign keys helps, e.g.

https://github.com/lxc/lxd/blob/582927233061c8e40002708f1e2d7a618ab149dc/lxd/db/cluster/update.go#L860-L861

**Comment by markylaing** [comment]
> @markylaing if you're finding that you're having to drop tables because of references to the table you have to remove and re-create to fix the description field, you may find that temporarily disabling foreign keys helps, e.g.
> 
> https://github.com/lxc/lxd/blob/582927233061c8e40002708f1e2d7a618ab149dc/lxd/db/cluster/update.go#L860-L861

@tomponline The `PRAGMA legacy_alter_table` command helps to get past the initial `main.projects table not found` error but unfortunately the `TestImportPreClusteringData` test is still failing because it expects to find networks that aren't there.

I think this might be because the `foreign_keys` PRAGMA [is a no-op within a transaction](https://sqlite.org/pragma.html#pragma_foreign_keys) (though dqlite might be different?). For the storage volumes, all data was re-added programatically even though the foreign keys were supposedly off: https://github.com/lxc/lxd/blob/582927233061c8e40002708f1e2d7a618ab149dc/lxd/db/cluster/update.go#L888-L913.

Since nodes and projects are linked by foreign keys to almost every table, I'm not sure there is a way around recreating almost the entire schema within this migration. That said, I can see there are already a lot of utilities for dumping and rebuilding the db, perhaps they are of use?

**Comment by tomponline** [comment]
@markylaing ok its a bit frustrating, but guess we'll have to recreate it fully. Thanks

**Comment by tomponline** [comment]
This will be good to get done before the 5.0 LTS release in April so we can clean up a DB quirk before the schema change freeze :)

File: lxd/db/cluster/schema.go [hunk 1]
```diff
@@ -155,79 +153,71 @@ CREATE TABLE "instances_profiles" (
     profile_id INTEGER NOT NULL,
     apply_order INTEGER NOT NULL default 0,
     UNIQUE (instance_id, profile_id),
-    FOREIGN KEY (instance_id) REFERENCES "instances"(id) ON DELETE CASCADE,
-    FOREIGN KEY (profile_id) REFERENCES profiles(id) ON DELETE CASCADE
-);
-CREATE INDEX instances_project_id_and_name_idx ON instances (project_id,
```
Comment (tomponline) [inline lxd/db/cluster/schema.go:161]: Whats happened here? The indexes appear to have been removed?

**Review by tomponline — CHANGES_REQUESTED** [review]
Seems like some indexes have been removed?

File: lxd/db/cluster/schema.go (hunk 1 above)
Comment (markylaing) [reply lxd/db/cluster/schema.go:161]: :grimacing: 

They get dropped when the tables are dropped. Thought I had recreated them all but clearly I have missed a few.

Comment (markylaing) [reply lxd/db/cluster/schema.go:161]: Fixed

**Review by tomponline — APPROVED** [review]
Thanks!

**Pull request merged**